plugins/rikki/heroeslounge/classes/discord/Attendance.php

```diff
@@ -9,55 +8,55 @@ class Attendance
-                                
-        while (count($jsonData) >= 1000 || $jsonData == null) {
-            $url = 'https://discordapp.com/api/guilds/200267155479068672/members';
-            if (count($jsonData) == 1000) {
-                $urlData = http_build_query(array("limit" => 1000, "after" => $jsonData[999]["user"]["id"]));
-            } else {
-                $urlData = http_build_query(array("limit" => 1000));
-            }
+      $presentUsers = [];
+      $jsonData = null;
 
-            $auth_header = AuthCode::getCode();
-            
-            $headers = [
-                "Content-Type:application/x-www-form-urlencoded",
-                $auth_header,
-                "User-Agent: HeroesLounge (http://heroeslounge.gg, 0.1)"
-            ];
+      while (count($jsonData) >= 1000 || $jsonData == null) {
+          $url = 'https://discordapp.com/api/guilds/200267155479068672/members';
+          if (count($jsonData) == 1000) {
+              $urlData = http_build_query(array("limit" => 1000, "after" => $jsonData[999]["user"]["id"]));
+          } else {
+              $urlData = http_build_query(array("limit" => 1000));
+          }
 
-            $ch = curl_init($url . "?" . $urlData);
-            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-            curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+          $auth_header = AuthCode::getCode();
 
-            $output = curl_exec($ch);
+          $headers = [
+              "Content-Type:application/x-www-form-urlencoded",
+              $auth_header,
+              "User-Agent: HeroesLounge (http://heroeslounge.gg, 0.1)"
+          ];
 
-            if (curl_errno($ch)) {
-                return false;
-            }
+          $ch = curl_init($url . "?" . $urlData);
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+          curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
 
-            $jsonData = json_decode($output, true);
+          $output = curl_exec($ch);
 
-            curl_close($ch);
+          if (curl_errno($ch)) {
+              return false;
+          }
 
-                            
-            foreach ($jsonData as $user) {
-                if (isset($user["user"])) {
-                    $presentDiscordTags[] = $user["user"]["username"] . "#" . $user["user"]["discriminator"];
-                }
-            }
-        }
+          $jsonData = json_decode($output, true);
 
-        return $presentDiscordTags;
+          curl_close($ch);
+
+
+          foreach ($jsonData as $user) {
+              if (isset($user["user"])) {
+                  $presentUsers[] = $user["user"];
+              }
+          }
+      }
+
+      return $presentUsers;
     }
 
     public function CheckAttendance()
     {
         $discordTags = Db::table("rikki_heroeslounge_sloths")->where("is_captain", 1)->whereIn('team_id', Db::table('rikki_heroeslounge_team_division')->whereIn('div_id', [1,2,3,4,5])->lists('team_id'))->lists("discord_tag");
```

Comment:
This is both bad style (do not use Db::table, instead go for Sloth::where("is_captain", 1)->... and also wrong, as it would only get you captains of teams that were in Div 1 through 5 in Season 4 - do not hardcode the division ids.
I am also unsure what this function is supposed to achieve - get attendance of all captains of all teams in the current season? Rather check for all teams of all divisions of all currently active seasons.

Reply:
I'm unsure a lot of the code in the Attendance.php file was written by Epixors at the time. I'm unsure if he also wrote this function as well

Reply:
I do not even know if this function is being used anymore, or by what. Might just be some legacy code lying around.